.github/workflows/python-packaging.yml

```diff
@@ -43,50 +42,37 @@ jobs:
   # for important transitional context.
   wheels:
     # Note: tries all supported Python versions as specified in apis/python/setup.py
-    name: Build wheels on ${{ matrix.os }}
+    name: Build ${{ matrix.python-version }}-${{ matrix.wheel-name }} wheel
     needs: sdist
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
+        python-version: [ '38', '39', '310', '311' ]
+        build: [ manylinux_x86_64, macosx_x86_64, macosx_arm64 ]
         include:
-          - os: ubuntu-20.04
-            cibw_build: 'cp3*-manylinux_x86_64'
-            platform: manylinux2014
-            wheel_name: manylinux2014
-          - os: macos-12
-            cibw_build: 'cp3*-macosx_x86_64'
+          - build: manylinux_x86_64
+            os: ubuntu-20.04
+            wheel-name: manylinux2014
+          - build: macosx_x86_64
+            os: macos-12
             cibw_archs_macos: x86_64
-            platform: macosx
-            wheel_name: macos-x86_64
-          - os: macos-14
-            cibw_build: 'cp3*-macosx_arm64'
+            wheel-name: macos-x86_64
+          - build: macosx_arm64
+            os: macos-12
             cibw_archs_macos: arm64
-            platform: macosx
-            wheel_name: macos-arm64
+            wheel-name: macos-arm64
     steps:
       - name: Download sdist artifact
-        uses: actions/download-artifact@v3
+        uses: actions/download-artifact@v4
         with:
           name: sdist
-      - name: rename sdist
+      - name: Rename sdist
         run: cp tiledbsoma-*.tar.gz tiledbsoma.tar.gz && ls -lh
-      # This is crucial for ongoing debug (do not remove it) as this shows the
```

Comment:
> This is crucial for ongoing debug (do not remove it)

Please do not remove this helpful information, for future use. It takes up a few dozen bytes of log output and I want it retained.